xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -265,19 +262,10 @@ func (b *clusterImplBalancer) UpdateClientConnState(s balancer.ClientConnState)
 }
 
 func (b *clusterImplBalancer) ResolverError(err error) {
```

Comment:
This method is also part of the `balancer.Balancer` interface and is invoked by gRPC and needs to be run inside of a serializer callback.

```
func (b *clusterImplBalancer) ResolverError(err error) {
	b.serializer.TrySchedule(func(context.Context) {
		b.child.ResolverError(err)
	})
}
```